docs/dev/guidelines/client.rst

```diff
@@ -35,11 +42,46 @@ Some general aspects:
 
 1. Do not export types/functions unless you need to share it across multiple components.
 2. Do not introduce new types/values to the global namespace.
-3. Shared types should be defined in 'types.ts'.
+3. Shared types/interfaces should be defined in 'types.ts'.
 4. Within a file, type definitions should come first.
 5. Interfaces and types offer almost the same functionality. To ensure consistency, choose ``interface`` over ``type`` whenever possible.
+
+.. code-block:: ts
+
+    // Dont do
+    type AngularLink = {
+        text: string;
+        routerLink: (string | number) [];
+    }
+
+    // Do
+    interface AngularLink {
+        text: string;
+        routerLink: (string | number) [];
+    }
+
+    // And this is also allowed (because interface is not possible here)
+    type RouterLinkPart = string | number;
+
+
+
 6. Use strict typing to avoid type errors: Do not use ``any``.
 
+7. Do not use anonymous data structures.
+
+.. code-block:: ts
+
+    interface AngularLink {
+        text: string;
+        routerLink: string [];
```

Comment:
```suggestion
        routerLink: string;
```

Reply:
an angular link consists of an array of strings, so I would like to keep it like this.

Reply:
Resolved in slack conversation